### DISCUSSION TITLE
Using MVT Style specification to style MVT layer in deck.gl

### DISCUSSION THREAD

**Discussion by corrigancd** [visgl/deck.gl#6892] [Q&A] [answered]
Hello deck.gl,
I asked a [question on stackoverflow](https://stackoverflow.com/questions/72140284/using-mvt-style-specification-to-style-mvt-layer-in-deck-gl). However, after exhausting online resources including the deck.gl documentation, I'm feeling that what I'm asking isn't currently possible. I'm hoping to confirm this is the case and if there are any plans to include this in deck.gl. I've pasted the question below for convenience: 

_I'm working on implementing an_ [MVT/PBF server](https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_v2/VectorTileServer/tile/%7Bz%7D/%7By%7D/%7Bx%7D.pbf) on _a_ [deck.gl map](https://deck.gl/) _and trying to find a way to implement the_ [accompanying style JSON](https://basemaps.arcgis.com/arcgis/rest/services/World_Basemap_v2/VectorTileServer/resources/styles), _i.e._ [Mapbox GL JS Style Specification file](https://docs.mapbox.com/mapbox-gl-js/style-spec/root/) _for this MVT/PBF server._

_I notice there is no style prop for the_ [deck.gl MVT layer](https://deck.gl/docs/api-reference/geo-layers/mvt-layer). _I could proceed by using the renderSubLayers prop to manually parse the style layer and match up the feature types and styles, but it would be tricky to implement the entire specification including max/min zoom visibility, filtering, text-halos etc._

_Hoping someone has a way of doing this?_

**Comment by Pessimistress** [comment, marked as answer]
deck.gl does not implement the Mapbox style spec. The library is better suited if you have complex styling logic or dynamic data. You can always use mapbox-gl to render a self-hosted MVT dataset and deck.gl as an overlay on top.

**Reply by corrigancd** [reply]
Thanks for the quick response and for providing a solution. I understand that `mapbox-gl` requires a Mapbox access token? If so, using it is not an option in my case.

Considering that Vector Tiles frequently have associated styling, they are becoming more widely used and there already is an [`MVT` deck.gl layer ](https://deck.gl/docs/api-reference/geo-layers/mvt-layer) type, is adding an argument to pass a url for a corresponding Mapbox [style JSON](https://docs.mapbox.com/mapbox-gl-js/style-spec/) something that might ever be considered for implementing directly in deck.gl?

**Reply by Pessimistress** [reply]
There have been discussions around this feature, but we don't have a timeline. There will need to be new layers written. Committing to support an evolving spec like this is also a lot of future maintenance work.

Regarding tokens - you can use [maplibre](https://maplibre.org/maplibre-gl-js-docs/api/) which is a free alternative.

**Comment by danloiterton** [comment]
Just FYI, I stumbled across deck.gl in my research, while attempting to find a way to render MVT tile layers in the google maps javascript API. It seemed like this was the ideal tool to meet this need, but alas, no styling!
